Add a project skeleton

connorkuehl opened the pull request:
This PR provides a basic makefile with a default target and a `clean` (`make clean`) target.

I've been building the project in a chroot like this:

```
$ schroot -c chroot:eoan-arm64 -- make
```

The Makefile doesn't modify the `CC` environment variable, so as long as your `CC` and `LD` environment variables are pointing at the right binaries it should work... I think.

I've been running it like this:

```
$ qemu-system-aarch64 -machine raspi3 -serial null -serial stdio -kernel kernel.elf
```

Commit:
Add basic .gitignore

Commit:
Add Makefile

Commit:
Implement initial bootable kernel skeleton

connorkuehl commented on start.S:
```diff
@@ -0,0 +1,62 @@
+	 * have cleared the entire section.
+	 */
+	cbz     w2, enter_kernel
+	str     xzr, [x1], #8
```
Does this work or is this just writing a zero to the same place for this entire loop?

squidboylan replied:
I think you're right and this just writes zero to the same place, i think if we swapped the `sub` and `str` and changed the `str` to be `str     xzr, [x1], w2` or something like that it'll do the right thing

connorkuehl replied:
https://github.com/bztsrc/raspi3-tutorial/issues/58#issuecomment-538686629

squidboylan commented on linker.ld:
```diff
@@ -20,4 +20,4 @@ SECTIONS
    /DISCARD/ : { *(.comment) *(.gnu*) *(.note*) *(.eh_frame*) }
 }
-__bss_size = (__bss_end - __bss_start)>>3;
+__bss_size = (__bss_end - __bss_start)>>3; /* size of bss in BYTES */
```
wait if this is in bytes, that makes it sounds like bss_end and bss_start are in bits?

squidboylan commented:
This LGTM, I'm happy to merge this, I have my docker stuff to add once this is in

Commit:
Add qemu{-gdb} targets

To run the kernel in a QEMU virtual machine, just run:

	$ make qemu

To run the kernel AND ask QEMU to expose a debugging port for GDB, run:

	$ make qemu-gdb

Then, in another session, run:

	$ gdb kernel.elf
	$ target remote localhost:1234

Also tidy up the Makefile just a little bit.
